test/filter.coffee

```diff
@@ -294,6 +299,110 @@ testFilter(
 	"a/b eq ((c ge d) and (e le f))"
 )
 
+
+# Test raw arrays
+testFilter(
+	$raw: [
+		"a/b eq $1 and a eq $2"
+		'c'
+		'd'
+	]
+	"a/b eq ('c') and a eq ('d')"
+)
+
+testFilter(
+	$raw: [
+		true
+	]
+	new Error('First element of array for $raw must be a string, got: boolean')
+)
+
+testFilter(
+	$raw: [
+		"a/b eq $1 and a eq $2"
+		{ c: 'd' }
+		$add: [
+			1
+			2
+		]
+	]
+	"a/b eq (c eq 'd') and a eq (((1) add (2)))"
+)
+
+testFilter(
+	$raw: [
+		"a/b eq $1"
+		$raw: '$$'
```

Comment:
This is to test a `$raw` inside a `$raw`, if you used just:
```coffee
$raw: [
	"a/b eq $1"
	'$$'
]
```
then it should end up as `a/b eq ('%24%24')` - ie have it escaped as a string rather than raw